contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1748,6 +1764,60 @@ pltsql_replace_non_determinstic(text *src_text, text *from_text, text *to_text,
 	return false;
 }
 
+/* 
+ * We are processesing USE DB command
+ * Communicate the same to common extension
+ * so that collation related information gets updated
+ */ 
+void
+tsql_set_db_collation(Oid database_collation_oid)
+{
+	/* Initialise collation callbacks */
+	init_and_check_collation_callbacks();
+
+	(*collation_callbacks_ptr->set_db_collation) (database_collation_oid);
+	return;
+}
+
+/* Find the collation corresponding to a specific database */
+char*
+get_collation_name_for_db(const char* dbname)
+{
+	HeapTuple	tuple;
+	Form_sysdatabases sysdb;
+	char *collation_name;
+
+	tuple = SearchSysCache1(SYSDATABASENAME, PointerGetDatum(cstring_to_text(dbname)));
+
+	if (!HeapTupleIsValid(tuple))
+		return NULL;
+
+	sysdb = ((Form_sysdatabases) GETSTRUCT(tuple));
+	collation_name = pstrdup(NameStr(sysdb->default_collation));
+
+	ReleaseSysCache(tuple);
+	return collation_name;
+}
+
+/* 
+ * We need to communicate to common extension
+ * that user has invoked USE DB command
+ * Hence, we need to update the cache related to -
+ * 1. database collation oid
+ * 2. database collation index
+ */
+void
+set_db_collation_internal(const char *db_name)
+{
+	Oid database_collation_oid;
+
+	/* Get collation oid corresponding to collation name */
+	database_collation_oid = get_collation_oid(list_make1(makeString((char*)get_collation_name_for_db(db_name))), false);
+
+	tsql_set_db_collation(database_collation_oid);
```

Comment:
creating another function `tsql_set_db_collation` does not seem necessary.

Reply:
Updated